pkg/systemd/quadlet/quadlet.go

```diff
@@ -509,10 +513,10 @@ func ConvertContainer(container *parser.UnitFile, isUser bool, unitsInfoMap map[
 	// Add a dependency on network-online.target so the image pull does not happen
 	// before network is ready
 	// https://github.com/containers/podman/issues/21873
-	// Prepend the lines, so the user-provided values
-	// override the default ones.
-	service.PrependUnitLine(UnitGroup, "After", "network-online.target")
-	service.PrependUnitLine(UnitGroup, "Wants", "network-online.target")
+	if service.LookupBooleanWithDefault(ContainerGroup, KeyQuadletDefaultDependencies, true) {
```

Comment:
Do we really want to add these generic keys to each group? Overall this just adds a ton of duplication in the docs.
And every time we add such a generic option we have to do that over and over again which does not look sane to me.

So maybe is it time to define a `[Quadlet]` section and then just name the key `DefaultDependencies` under there? This would only need to be documented once and then should work for all units. @ygalblum WDYT?

Reply:
I had a similar thought but didn't consider the long-term impact on new features. +1 from my side.

Reply:
I agree and I was actually thinking about it for a while now. For example, I would move `ContainersConfModule` to there as well (I know, for backward compatibility we'll need to support both)
I'm just not sure we want to force @lambinoo to make this change.
Let's hold this PR for a sec and I'll try to find some time to add this section

Reply:
I don't mind. Just pushed the (what I think are) the required changes for this!